src/components/tooltip/bl-tooltip.ts

```diff
@@ -0,0 +1,119 @@
+import { CSSResultGroup, html, LitElement, TemplateResult } from 'lit';
+import { customElement, property, query, state } from 'lit/decorators.js';
+import { computePosition, flip, shift, offset, arrow, Strategy } from '@floating-ui/dom';
+import { classMap } from 'lit/directives/class-map.js';
+import { ReferenceElement } from '@floating-ui/core';
+import style from './bl-tooltip.css';
+
+export type Placement =
+  | 'top-start'
+  | 'top'
+  | 'top-end'
+  | 'bottom-start'
+  | 'bottom'
+  | 'bottom-end'
+  | 'left-start'
+  | 'left'
+  | 'left-end'
+  | 'right-start'
+  | 'right'
+  | 'right-end';
+
+  /**
+ * @tag bl-tooltip
+ * @summary Baklava Tooltip component
+ *
+ * @property {string} placement - Sets the tooltip placement
+ *
+ * @cssproperty --bl-tooltip-position - Sets the position. Default value is 'absolute'
+ */
+
+@customElement('bl-tooltip')
+export default class BlTooltip extends LitElement {
+  static get styles(): CSSResultGroup {
+    return [style];
+  }
+
+  @query('.tooltip') private tooltip: HTMLElement;
+  @query('.trigger') private trigger: ReferenceElement;
+  @query('.arrow') private arrow: HTMLElement;
+
+  /**
+   * Sets placement of the tooltip
+  */
+  @property({ type: String })
+  placement: Placement = 'top';
+
+  @state() private _visible = false;
+  @state() private _position : Strategy = 'absolute';
+  @state() private host : HTMLElement;
+
+  connectedCallback() {
+    super.connectedCallback();
+
+    setTimeout(() => {
+     this.host = this.shadowRoot?.host as HTMLElement;
+     this._position = getComputedStyle(this.host).getPropertyValue('--bl-tooltip-position') as Strategy; 
+    });
+  }
+
+  private setTooltip() {
+    computePosition(this.trigger, this.tooltip, {
+      placement: this.placement,
+      strategy: this._position,
+      middleware: [offset(8), shift({ padding: 5 }), flip(),arrow({ element: this.arrow,padding: 5})],
+    }).then(({ x, y, placement, middlewareData }) => {
+      const arrowX = middlewareData.arrow?.x != null ? `${middlewareData.arrow?.x}px` : ' ';
+      const arrowY = middlewareData.arrow?.y != null ? `${middlewareData.arrow?.y}px`  : ' ';
```

Comment:
Is there a chance to have a `null` value for those values? If so what happens when we set empty string for `--bl-tooltip-arrow-left` ?

Reply:
The arrow middleware is designed only to position the arrow on one axis. If the tooltip placement is `top` or `bottom` it gives only `x` position and if the tooltip placement is `right` or `left` it gives only `y` position. 

We set an empty string because we don't want to define `tooltip-arrow-left` when the placement is `right` or `left`. 
And we don't want to define `tooltip-arrow-top` when the placement is `top` or `bottom`.